radix-engine/src/system/system.rs

```diff
@@ -468,6 +477,15 @@ where
                         )));
                     }
 
+                    self.api
+                        .kernel_get_callback()
+                        .modules
+                        .events
+                        .add_replacement(
+                            (node_id, ObjectModuleId::SELF),
```

Comment:
Out of interest- what are these three lines for?

You seem to be mapping from `SELF` to `AccessRules` but I imagine you should be mapping from `AccessRules` to `AccessRules`? But an internal node doesn't have access rules (or indeed any module other than self) so this wouldn't make sense.

Relatedly, I'd perhaps expect you to just provide a (node_id => address) mapping (ignoring the module id) and map independently of the module id :)

Reply:
This is actually intentional.

So, a bit of a background on this bug and how it occurs. During the account (and identities) secure-ifying process an `AccessRules` node is created. This isn't a node module, no it's an actual node with a node id. 

During the secure-ifying process the "update rules" method is called on this `AccessRules` node which emits an event. This event is then stored in the event store as: "The SELF node module of this AccessRules node emitted this event". It's stored in the event store in this way because at this point this `AccessRules` is a node and not a node module, so we can't associate it with any parent or anything of that sort.

When the parent node (the account) is globalized, all of the nodes associated with the node modules are dropped and the associated substates are stored as node modules under the parent node. So, in the case of the example I provided above, the `AccessRules` node is dropped and its substates are used for the `AccessRules` node module of the account. 

However, the event store still has the event that I mentioned earlier, the "The SELF node module of this AccessRules node emitted this event", but this node has already been dropped and thus its schema can not be looked up. 

So, to fix this, when a node is globalized we replace all of the events emitted by the SELF node module of the node modules to the appropriate parent node module. So, in this case, we're trying to replace all events emitted by the SELF node module of the AccessRules node to say that they've been emitted by the AccessRules node module of the parent node (the account).